### PR TITLE
Generate Lower Bound Conditions for issue 1014

### PR DESCRIPTION
**Pull request by kun-zh**
Generate Lower Bound Conditions for issue #1014

### CODE REVIEW

File: src/schedule/message_passing.cc [hunk 1]
```diff
@@ -477,7 +477,11 @@ std::vector<Expr> MakeBoundCheck(
     CHECK(iv->dom.defined());
     if (!skip_ivar_domain && !iv->dom.same_as(dom)) {
       Expr value = ComputeExpr<Sub>(value_map.at(iv), iv->dom->min);
+      Expr vmin = EvalSet(value, iset_dmap).min();
       Expr vmax = EvalSet(value, iset_dmap).max();
+      if (vmin.type() != value.type() || !can_prove(vmin > (iv->dom->min - 1))) {
```
Comment (tqchen) [inline src/schedule/message_passing.cc]: vmin >= iv->dom->min

File: src/schedule/message_passing.cc [hunk 2]
```diff
@@ -477,7 +477,11 @@ std::vector<Expr> MakeBoundCheck(
...
+        preds.emplace_back(value > (iv->dom->min - 1));
```
Comment (tqchen) [inline src/schedule/message_passing.cc]: value >= iv->dom->min

**Comment by tqchen** [comment]
there is a regression test failure, @kun-zh can you look into that? http://mode-gpu.cs.washington.edu:8080/blue/organizations/jenkins/dmlc%2Ftvm/detail/PR-1091/3/pipeline

**Comment by kun-zh** [comment]
refined and fixed regression failure

File: src/schedule/message_passing.cc [hunk 3]
```diff
@@ -477,9 +477,13 @@ std::vector<Expr> MakeBoundCheck(
     CHECK(iv->dom.defined());
     if (!skip_ivar_domain && !iv->dom.same_as(dom)) {
       Expr value = ComputeExpr<Sub>(value_map.at(iv), iv->dom->min);
+      Expr vmin = EvalSet(value, iset_dmap).min();
```
Comment (tqchen) [inline src/schedule/message_passing.cc]: Do EvalSet once, instead of multiple times

Comment (tqchen) [reply src/schedule/message_passing.cc]: specifically, cache the result of EvalSet and then get min and max

Comment (kun-zh) [reply src/schedule/message_passing.cc]: done

**Pull request merged**